### DISCUSSION TITLE
Akai MPK Mini MK3 not working

### DISCUSSION THREAD

**Discussion by koffo** [probonopd/MiniDexed#355] [Q&A]
Hi everyone,

first of all thanks for this project, it's great having the glorious DX7 packed up in a small box!

I'm having issues with the Akai MPK Mini MK3 midi controller. According to producer specs it should be a class-compliant, USB-MIDI keyboard. Anyways, when I try to connect it to the RPI it doesn't produce any sound.
What I tried to do is:

- Set usbspeed=full
- Set usbignore for all the interfaces but 1-3-0

Here is the video output I have, it is evident that there are several interfaces being recognized:

![20220919_214635](https://user-images.githubusercontent.com/62171109/191105911-93f6868a-9e7b-495e-a0b9-d56861d83cee.jpg)

Any ideas on what to do?

Thanks and regards,
Andrea

**Comment by probonopd** [comment]
I don't have this hardware but here someone tested the Akai MPK mini over USB successfully:
https://github.com/probonopd/MiniDexed/discussions/279#discussioncomment-2958215

Don't know how different the MK3 is.

**Reply by koffo** [reply]
Ok, I'm checking if there is something special to be done. Thanks for answering!

**Reply by koffo** [reply]
Finally I found a way out: the problem is something related to the RPI 4.
I changed the board to an old RPI 3B I used for an old project and, without changing anything else, it started working.
Now I have a fully functional AKAI MPK Mini MK3!

**Reply by probonopd** [reply]
@rsta2 can you imagine why the AKAI MPK Mini MK3 might be working on the RPi 3 but not on the RPi 4?

**Reply by Gustavomurta** [reply]
The Akai MPK mini is powered via USB. The Raspberry Pi4 consumes much more power than the RPI3. It may be that the power supply via USB is not supporting the consumption of the RPI4 + Akai MPK mini.

To avoid this kind of problem, I adapted the USB connection with my Akai MPK mini. I separated the 5V from the USB port of the RPI. In this case, Akai is powered through the 5V regulator.

![MiniDexed amplifier](https://github.com/probonopd/MiniDexed/assets/4587366/e4aca385-064f-44bd-96e9-e2922396ea40)


![miniDexed RaspPi Zero](https://github.com/probonopd/MiniDexed/assets/4587366/b781b19e-9e3c-4f4b-87ff-675c4104c796)

**Comment by Gustavomurta** [comment]
**My USB adaptor - MIDI KBD**

The 5V pin (V BUS) are not interconnected.
The other pins are interconnected (D+,D- and GND). 
The 5V for the MIDI keyboard comes from the LM2596 Regulator.
Use a USB extender cable at left. At right connect the MIDI KBD. 

**Attention ! Set the LM2596 Regulator output to 5V before connecting any device. Or you could burn some.**
Or you can use another regulator with a fixed output of 5V (but with current >= 3 Amps).

![USB Midi adaptador](https://github.com/probonopd/MiniDexed/assets/4587366/9fada189-967f-4767-bd62-b826fa8ad9d8)
